drivers/input/input_gt911.c

```diff
@@ -130,29 +133,36 @@ static int gt911_process(const struct device *dev)
 		return r;
 	}
 
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
+	/*
+	 * obtain point X, Y coordinates and event
 	 */
-	reg_addr = REG_FIRST_POINT;
-	r = gt911_i2c_write_read(dev, &reg_addr, sizeof(reg_addr), &point_reg, sizeof(point_reg));
-	if (r < 0) {
-		return r;
+	if (points) {
+		for (uint8_t i = 0; i < points; i++) {
+			reg_addr = REG_POINT_ADDR(i);
+
+			r = gt911_i2c_write_read(dev, &reg_addr, sizeof(reg_addr), &point_reg,
+						 sizeof(point_reg));
+			if (r < 0) {
+				return r;
+			}
+
+			row = ((point_reg.high_y) << 8U) | point_reg.low_y;
+			col = ((point_reg.high_x) << 8U) | point_reg.low_x;
+			LOG_DBG("touch : %u, row: %u, col: %u", i, row, col);
+#ifdef CONFIG_INPUT_GT911_MULTITOUCH_ENABLED
+			input_report_abs(dev, INPUT_ABS_MT_SLOT, i, true, K_FOREVER);
+#endif
+			input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
+			input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
+			input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+		}
 	}
 
-	pressed = (points == 1);
-	row = ((point_reg.high_y) << 8U) | point_reg.low_y;
-	col = ((point_reg.high_x) << 8U) | point_reg.low_x;
-
-	LOG_DBG("pressed: %d, row: %d, col: %d", pressed, row, col);
-
-	if (pressed) {
-		input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
-		input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
-		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
-	} else {
+	if (prev_points > points) {
 		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
 	}
```

Comment:
@fabiobaltieri : I think this should be outside of the loop because if we put it inside, when points decrease to 0, we couldn't detect the touch-off event.

BTW, I fixed the logic errors when

- reporting the touch-off event : when multitouch is not enabled, this event should be reported only onces when points = 0
- reporting the touch events :  when multitouch is not enabled, touch events should be reported only for the 1st point

Hope this is OK this time :)

Reply:
sorry I'm having a hard time following: when MT is enabled you should be able to detect touch on and touch off for every tracked finger, so let's say you keep pressing on one finger and you add a second one you'd get a set of <mt slot 1><touch on> and then when it releases <mt slot 1><touch off> and eventually when the other finger releases you should get <mt slot 0><touch off>. Is that what's this is doing now? I can't imagine it being the case, if you had multiple presses on the device and release all of them at once I'd expect a series of <slot x><touch off> so it must be in the cycle.

Reply:
@fabiobaltieri, can you take a look again please? the points are now individually tracked through their ID, coordinates and their state (press/release)